src/HttpServer.php

```diff
@@ -189,11 +190,23 @@ private function handlePromise(Connection $connection, Promise $promise)
         $promise->then(
             function ($response) use ($connection, $promise){
                 $responseString = RingCentral\Psr7\str(new Response(500));
-
                 if ($response instanceof Response) {
+                    $body = $response->getBody();
+                    if ($body instanceof ReadableStreamInterface) {
+                        $headerResponse = new Response(
```

Comment:
Create a response object to send the header at the beginning. The body will be streamed.

I found no possibilities to get just the header information as a valid HTTP string to send it to the client.

Any ideas?